src/System.Linq.Expressions/src/System/Linq/Expressions/Compiler/ILGen.cs

```diff
@@ -96,52 +96,46 @@ internal static void EmitLoadValueIndirect(this ILGenerator il, Type type)
         {
             ContractUtils.RequiresNotNull(type, nameof(type));
 
-            if (type.GetTypeInfo().IsValueType)
+            switch (type.GetTypeCode())
```

Comment:
There's something peculiar about many methods in `ILGen` that analyze the `TypeCode`. Most seem to special-case for `IsEnum` and emit the generic opcode for value types in that case (e.g. `ldobj` here), rather than the specialized opcode for the underlying type. This is unlike the C# compiler which emits e.g. `ldind.i4` for an enum such as `ConsoleColor`; see e.g. `Microsoft.CodeAnalysis.CSharp.CodeGen.CodeGenerator.EmitExpression.EmitLoadIndirect` which does a similar thing as `GetTypeCode`, namely use the `TypeCode` for the enum's underlying type.